x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -195,6 +199,12 @@ async function createSetupSideEffects(
   logger.debug('Setting up Space settings');
   await ensureSpaceSettings(appContextService.getConfig()?.spaceSettings ?? []);
 
+  logger.debug('Setting up delete unenrolled agents setting');
```

Comment:
I tried to enable it then disable it via the kibana config and it stay enabled, it is expected? in my opinion if that settings is managed via the kibana config it should probably be the only source of truth 
```
xpack.fleet.enableDeleteUnenrolledAgents: false
```

Reply:
Good catch, it was a bug, it is fixed now.

Reply:
I agree that it's clearer to have one single source of truth. Are we ok to keep this setting in preconfiguration only, and remove from the UI @kpollich @nimarezainia ?

Reply:
I think, similar to most of the configuration we should allow to configure it both from UI or kibana config, but if it's configured in preconfiguration this should become the only way to configure it (until it's removed from the config)

Reply:
In that case I have to introduce an `is_preconfigured` flag to know if the delete flag was preconfigured.